moveit_core/robot_model/src/robot_model.cpp

```diff
@@ -968,35 +968,37 @@ JointModel* RobotModel::constructJointModel(const urdf::Joint* urdf_joint, const
   {
     new_joint_model->setDistanceFactor(new_joint_model->getStateSpaceDimension());
     const std::vector<srdf::Model::PassiveJoint>& pjoints = srdf_model.getPassiveJoints();
+    std::string joint_name = new_joint_model->getName();
     for (const srdf::Model::PassiveJoint& pjoint : pjoints)
     {
-      if (new_joint_model->getName() == pjoint.name_)
+      if (joint_name == pjoint.name_)
       {
         new_joint_model->setPassive(true);
         break;
       }
     }
-    for (const srdf::Model::JointProperty& property : srdf_model.getJointProperties(new_joint_model->getName()))
+    for (const auto& property : srdf_model.getJointProperties(joint_name))
     {
-      if (property.property_name_ == "angular_distance_weight")
+      std::string property_name = property.first;
+      std::string property_value = property.second;
+      if (property_name == "angular_distance_weight")
       {
         double angular_distance_weight;
         try
         {
           std::string::size_type sz;
-          angular_distance_weight = std::stod(property.value_, &sz);
-          if (sz != property.value_.size())
+          angular_distance_weight = std::stod(property_value, &sz);
```

Comment:
This parsing of double values is repeated below. Better move into a utility function!

Reply:
`std::stod` is 'designed' for use with natural language and assumes localized decimal separators. That's the polite version :wink:
Please use the conversion helpers in `moveit::tools`

Can we have CI ban using these functions for good?

Reply:
Thanks for the feedback! 

So far, I've refactored the code to use `moveit::core::toDouble` instead of `std::stod` [here](https://github.com/scchow/moveit/compare/dev/diff-drive...scchow:moveit:dev/robot-str-parse), which has signficantly simplified the double parsing.

I was wondering whether a separate utility function is still necessary, since it is now just calling `toDouble` with a catch for improper values.

Reply:
This is already a good improvement. Nevertheless, the error handling still constitutes a significant amount of redundancy. Simply define a lambda function parsing a specific double property with error handling.